src/List/Transformer.hs

```diff
@@ -420,6 +428,81 @@ select = Data.Foldable.foldr cons empty
   where
     cons x xs = pure x <|> xs
 
+
+-- | @take n xs@ takes @n@ elements from the head of @xs@.
+--
+-- >>> let list = asum . map (\x -> liftIO (print (show x)) >> return x)
```

Comment:
These doctests fail on older versions of GHC due to the monomorphism restriction.  I think you can fix this by just adding `>>> :set -XNoMonomorphismRestriction` to the `$setup`